File: scala-hoots/src/main/java/io/hoots/match/domain/Matches.java
package io.hoots.match.domain;

import io.hoots.fingerprint.domain.HashMatch;

import javax.sound.sampled.AudioFormat;
import java.util.List;

/**
 * Created by rwadowski on 14.04.17.
 */
public class Matches {

    private final List<HashMatch> matchList;
    private final AudioFormat inputFormat;

    public Matches(List<HashMatch> matchList, AudioFormat inputFormat) {
        this.matchList = matchList;
        this.inputFormat = inputFormat;
    }

    public List<HashMatch> getMatchList() {
        return this.matchList;
    }

    public AudioFormat getInputFormat() {
        return this.inputFormat;
    }
}

File: scala-hoots/src/main/scala/io/hoots/results/domain/Matches.scala
package io.hoots.results.domain

import javax.sound.sampled.AudioFormat

import io.hoots.fingerprint.domain.{HashMatch, Point}
import io.hoots.input.domain.Item

/**
  * Created by rwadowski on 19.04.17.
  */
case class Matches(list: Seq[HashMatch],
                   audioFormat: AudioFormat) {

}


case class MatchResult(scores: Map[Item, Score],
                       matchPoints: Map[Item, Point],
                       audioFormat: AudioFormat) {

  lazy val matchTimestamps: Map[Item, Float] = {
    var map: Map[Item, Float] = Map.empty
    val sampleSizeInBytes = audioFormat.getSampleSizeInBits / 8
    for((item, point) <- matchPoints) {
      val chunk = point.chunk
      val ts = chunk.chunkByte / (audioFormat.getSampleRate * sampleSizeInBytes)
      map = map + (item -> ts)
    }
    map
  }
}